specs/Standards/MultiToken/README.md

```diff
@@ -0,0 +1,539 @@
+
+    /// Returns the total supply of the token in a decimal string representation given token_id.
+    fn mt_total_supply(&self, token_id: TokenId) -> U128;
+
+    // Returns the total supplies of the tokens given by token_ids in a decimal string representation.
+    fn mt_total_supply_batch(&self, token_ids: Vec<TokenId>) -> Vec<U128>;
+}
+```
+### Receiver Trait
+#### Notes
+- TokenId is of type String
+```
+pub trait MultiTokenReceiver {
+    /// Take some action after receiving a MultiToken-tokens token
+    ///
+    /// Requirements:
+    /// * Contract MUST restrict calls to this function to a set of whitelisted MultiToken
+    ///   contracts
+    ///
+    /// Arguments:
+    /// * `sender_id`: the sender of `mt_transfer_call`
+    /// * `previous_owner_id`: the account that owned the tokens prior to it being
+    ///   transferred to this contract, which can differ from `sender_id` if using
+    ///   Approval Management extension
+    /// * `token_ids`: the `token_ids` argument given to `mt_transfer_call`
+    /// * `msg`: information necessary for this contract to know how to process the
+    ///   request. This may include method names and/or arguments.
+    ///
+    /// Returns true if tokens should be returned to `sender_id`
+    fn mt_on_transfer(
+        &mut self,
+        sender_id: AccountId,
+        token_ids: Vec<TokenId>,
+        amounts: Vec<U128>,
+        msg: String,
+    ) -> PromiseOrValue<Vec<U128>>;
+}
+```
+### Resolver Trait
+#### Notes
+- TokenId is of type String
+```
+/// Used when MultiTokens are transferred using `mt_transfer_call`. This is the method that's called after `mt_on_transfer`. This trait is implemented on the MultiToken contract.
+pub trait MultiTokenResolver {
+    /// Finalize an `mt_transfer_call` chain of cross-contract calls.
+    ///
+    /// The `mt_transfer_call` process:
+    ///
+    /// 1. Sender calls `mt_transfer_call` on MultiToken contract
+    /// 2. MultiToken contract transfers token from sender to receiver
+    /// 3. MultiToken contract calls `mt_on_transfer` on receiver contract
+    /// 4+. [receiver contract may make other cross-contract calls]
+    /// N. MultiToken contract resolves promise chain with `mt_resolve_transfer`, and may
+    ///    transfer token back to sender
+    ///
+    /// Requirements:
+    /// * Contract MUST forbid calls to this function by any account except self
+    /// * If promise chain failed, contract MUST revert token transfer
+    /// * If promise chain resolves with `true`, contract MUST return token to
+    ///   `sender_id`
+    ///
+    /// Arguments:
+    /// * `previous_owner_id`: the owner prior to the call to `mt_transfer_call`
+    /// * `receiver_id`: the `receiver_id` argument given to `mt_transfer_call`
+    /// * `token_ids`: the `token_ids` argument given to `mt_transfer_call`
+    /// * `approvals`: if using Approval Management, contract MUST provide
```

Comment:
I see `approvals` referenced here, but it's not shown in the function signature below?  It's not clear from reading this what exactly is inside of this; we say it 'may be multiple sets of approvals' is it an array of arrays?